core/services/functions/external_adapter_client.go

```diff
@@ -62,14 +62,23 @@ type bridgeAccessor struct {
 var _ BridgeAccessor = (*bridgeAccessor)(nil)
 
 type requestPayload struct {
-	Endpoint            string       `json:"endpoint"`
-	RequestId           string       `json:"requestId"`
-	JobName             string       `json:"jobName"`
-	SubscriptionOwner   string       `json:"subscriptionOwner"`
-	SubscriptionId      uint64       `json:"subscriptionId"`
-	Flags               RequestFlags `json:"flags"` // marshalled as an array of numbers
-	NodeProvidedSecrets string       `json:"nodeProvidedSecrets"`
-	Data                *RequestData `json:"data"`
+	Endpoint            string           `json:"endpoint"`
+	RequestId           string           `json:"requestId"`
+	JobName             string           `json:"jobName"`
+	SubscriptionOwner   string           `json:"subscriptionOwner"`
+	SubscriptionId      uint64           `json:"subscriptionId"`
+	Flags               RequestFlags     `json:"flags"` // marshalled as an array of numbers
+	NodeProvidedSecrets string           `json:"nodeProvidedSecrets"`
+	Data                *computationData `json:"data"`
+}
+
+type computationData struct {
```

Comment:
Can we add omitempty to Secrets field in RequestData instead of creating a copy of the same struct?